ocaml/squeezed/lib/squeeze.ml

```diff
@@ -84,36 +84,33 @@ let domain_to_string_pairs (x : domain) =
   ; ("inaccuracy_kib", i64 x.inaccuracy_kib)
   ]
 
-module IntMap = Map.Make (struct
-  type t = int
+module DomainSet = Set.Make (struct
+  type t = domain
 
-  let compare = compare
+  let compare {domid= domid_a; _} {domid= domid_b; _} =
+    Int.compare domid_a domid_b
 end)
 
 (** Per-Host data *)
 type host = {
-    domains: domain list  (** VMs running on this host *)
-  ; domid_to_domain: domain IntMap.t  (** total free memory on this host *)
-  ; free_mem_kib: int64
+    domains: DomainSet.t  (** VMs running on this host *)
+  ; free_mem_kib: int64  (** total free memory on this host *)
 }
 
-let make_host ~domains ~free_mem_kib =
-  {
-    domains
-  ; domid_to_domain=
-      List.fold_left
-        (fun map domain -> IntMap.add domain.domid domain map)
-        IntMap.empty domains
-  ; free_mem_kib
-  }
+let make_host ~domains ~free_mem_kib = {domains; free_mem_kib}
 
 let string_pairs_to_string (x : (string * string) list) =
   String.concat "; " (List.map (fun (k, v) -> k ^ "=" ^ v) x)
 
 let domain_to_string d = string_pairs_to_string (domain_to_string_pairs d)
 
 let host_to_string_pairs (x : host) =
-  let domains = String.concat "; " (List.map domain_to_string x.domains) in
+  let domains =
+    DomainSet.fold
+      (fun domain acc -> domain_to_string domain :: acc)
+      x.domains []
```

Comment:
This returns the domains in reverse order, is this important?

Reply:
I have checked those code is unused anywhere. I decide to delete those code.